src/runtime_src/core/pcie/driver/linux/xocl/subdev/xgq.c

```diff
@@ -151,6 +163,8 @@ struct xocl_xgq_vmr {
 	struct semaphore 	xgq_data_sema;
 	struct semaphore 	xgq_log_page_sema;
 	struct xgq_cmd_cq_default_payload xgq_cq_payload;
+	//clock scaling structure to save all feature details
+	struct xgq_clk_scaling_info xgq_cs_struct;
```

Comment:
We don't have to keep extra storage for cs. The xgq_cq_payload has everything cached already.
Check vmr_status_show for a reference.
You actually have already casted to it in the clk_scaling_cq_result_copy.

Reply:
Hi @xdavidz thanks for review. I have taken care of your suggestions. Please review it again, thank you.